test/Result/Result.Tests.cs

```diff
@@ -29,5 +29,35 @@ public void Failure()
             Assert.False(result2);
             Assert.True(result2.IsError);
         }
+
+        [Fact]
+        public async Task SuccessWithTask()
+        {
+            // Arrange
+            Result result1 = Result.Ok();
+            Task<Result> taskResult() => Task.FromResult(result1);
+
+            // Act
+            var result2 = await taskResult();
+
+            //Assert
+            Assert.True(result2);
+            Assert.False(result2.IsError);
```

Comment:
Since we already testes Ok and Fail in the first two tests this assertion is not necessary.

Reply:
If these assertions are not necessary, what should I do to assert in this method?

I can only think about: `Assert.True(result);`

Reply:
That's it! ✌🏾